packages/page-accounts/src/modals/CreateEthDerivationPath.tsx

```diff
@@ -0,0 +1,147 @@
+// Copyright 2017-2021 @polkadot/app-accounts authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import BN from 'bn.js';
+import React, { useEffect, useRef, useState } from 'react';
+
+import { Checkbox, Dropdown, Input, InputNumber, MarkError, MarkWarning, Modal } from '@polkadot/react-components';
+
+import { useTranslation } from '../translate';
+import { DeriveValidationOutput, PairType } from '../types';
+
+interface Props {
+  className?: string;
+  onChange: (string: string) => void;
+  seedType: string;
+  derivePath: string;
+  deriveValidation: DeriveValidationOutput | undefined;
+  seed: string;
+  addressFromSeed: (seed: string, derivePath: string, pairType: PairType) => string;
+}
+
+export const ETH_DEFAULT_PATH = "m/44'/60'/0'/0/0";
+
+function CreateEthDerivationPath ({ addressFromSeed,
+  className,
+  derivePath,
+  deriveValidation,
+  onChange,
+  seed,
+  seedType }: Props): React.ReactElement<Props> {
+  const { t } = useTranslation();
+  const [addIndex, setAddIndex] = useState(0);
+  const [customIndex, setCustomIndex] = useState(0);
+  const addressList = new Array(5).fill(0).map((_, i) => ({
+    key: i,
+    text: t('Address index {{index}} - {{address}}', {
+      replace: { address: addressFromSeed(seed, `m/44'/60'/0'/0/${i}`, 'ethereum'), index: i }
```

Comment:
Would not suggest doing this up-front. This is a very costly operation.

Additionally, it gets done on each render, not only when the seed changes.